tests/test_requests_compat.py

```diff
@@ -0,0 +1,2049 @@
+
+    def test_fragment_maintained_on_redirect(self, httpbin):
+        fragment = "#view=edit&token=hunter2"
+        r = requests.get(httpbin("redirect-to?url=get") + fragment)
+
+        assert len(r.history) > 0
+        assert r.history[0].request.url == httpbin("redirect-to?url=get") + fragment
+        assert r.url == httpbin("get") + fragment
+
+    def test_HTTP_200_OK_GET_WITH_PARAMS(self, httpbin):
+        heads = {"User-agent": "Mozilla/5.0"}
+
+        r = requests.get(httpbin("user-agent"), headers=heads)
+
+        assert heads["User-agent"] in r.text
+        assert r.status_code == 200
+
+    def test_HTTP_200_OK_GET_WITH_MIXED_PARAMS(self, httpbin):
+        heads = {"User-agent": "Mozilla/5.0"}
+
+        r = requests.get(
+            httpbin("get") + "?test=true", params={"q": "test"}, headers=heads
+        )
+        assert r.status_code == 200
+
+    def test_set_cookie_on_301(self, httpbin):
+        s = requests.session()
+        url = httpbin("cookies/set?foo=bar")
+        s.get(url)
+        assert s.cookies["foo"] == "bar"
+
+    def test_cookie_sent_on_redirect(self, httpbin):
+        s = requests.session()
+        s.get(httpbin("cookies/set?foo=bar"))
+        r = s.get(httpbin("redirect/1"))  # redirects to httpbin('get')
+        assert "Cookie" in r.json()["headers"]
+
+    def test_cookie_removed_on_expire(self, httpbin):
+        s = requests.session()
+        s.get(httpbin("cookies/set?foo=bar"))
+        assert s.cookies["foo"] == "bar"
+        s.get(
+            httpbin("response-headers"),
+            params={"Set-Cookie": "foo=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT"},
+        )
+        assert "foo" not in s.cookies
+
+    @pytest.mark.xfail(
+        reason="""
+        assert '"\\"bar:baz\\""' == '"bar:baz"'
+        """
+    )
+    def test_cookie_quote_wrapped(self, httpbin):
+        s = requests.session()
+        s.get(httpbin('cookies/set?foo="bar:baz"'))
+        assert s.cookies["foo"] == '"bar:baz"'
+
+    def test_cookie_persists_via_api(self, httpbin):
+        s = requests.session()
+        r = s.get(httpbin("redirect/1"), cookies={"foo": "bar"})
+        assert "foo" in r.request.headers["Cookie"]
+        assert "foo" in r.history[0].request.headers["Cookie"]
+
+    def test_request_cookie_overrides_session_cookie(self, httpbin):
+        s = requests.session()
+        s.cookies["foo"] = "bar"
+        r = s.get(httpbin("cookies"), cookies={"foo": "baz"})
+        assert r.json()["cookies"]["foo"] == "baz"
+        # Session cookie should not be modified
+        assert s.cookies["foo"] == "bar"
+
+    def test_request_cookies_not_persisted(self, httpbin):
+        s = requests.session()
+        s.get(httpbin("cookies"), cookies={"foo": "baz"})
+        # Sending a request with cookies should not add cookies to the session
+        assert not s.cookies
+
+    @pytest.mark.xfail(reason="cookies object are different instances (but == would work)")
```

Comment:
Hmm... is there any value in the cookiejars being the same instance?

Reply:
Dunno. For now I’m doing this work pretty systematically. We can then go over the xfails and decide whether it’s a use case we want to support to/if it has value at all. :)

Reply:
Refactored this to skip the `is` check — I figured what we really tested there was whether cookies were sent correctly.